pkg/auth/multiauthenticator.go

```diff
@@ -0,0 +1,67 @@
+package auth
+
+import (
+	"context"
+	"fmt"
+	"strings"
+)
+
+type IssuerWithClientID struct {
+	Issuer   string // Issuer of the token.
+	ClientID string // ClientID of the token, can be empty.
+}
+
+type multiAuthenticatorWithClientIDImpl struct {
+	authenticators map[IssuerWithClientID]Authenticator
+}
+
+func NewMultiAuthenticatorWithClientID(
+	authenticators map[IssuerWithClientID]Authenticator) MultiAuthenticatorWithClientID {
+	return &multiAuthenticatorWithClientIDImpl{
```

Comment:
Should we create a new map to store authenticators and copy the contents into it?

nit: should we do some basic input validation? For example, is it ok to add same issuer once with empty and once with non-empty clientID? We may also want to prevent the same (issuer, clientID) combination multiple times. This is not auto-prevented due to the "contains" check.

Reply:
done